File: applications/luci-app-statistics/luasrc/model/cbi/luci_statistics/cpufreq.lua
-- Licensed to the public under the Apache License 2.0.

m = Map("collectd",
	translate("CPU Frequency Plugin Configuration"),
	translate("This plugin collects statistics about the processor frequency scaling."))

-- collectd_cpufreq config section
s = m:section( NamedSection, "cpufreq", "plugin" )

-- collectd_cpufreq.enable
enable = s:option( Flag, "enable", translate("Enable this plugin") )
enable.default = 0

-- collectd_cpufreq.extraitems
extraitems = s:option( Flag, "ExtraItems", translate("Extra items"), translate("More details about frequency usage and transitions"))
extraitems.default = "0"
extraitems.optional = true
extraitems:depends( "enable", 1 )

return m

File: applications/luci-app-statistics/htdocs/luci-static/resources/view/statistics/plugins/cpufreq.js
'use strict';
'require baseclass';
'require form';
'require statistics.pluginUtil as pluginUtil';

return baseclass.extend({
	title: _('CPU Frequency Plugin Configuration'),
	description: _('This plugin collects statistics about the processor frequency scaling.'),

	addFormOptions: function(s) {
		var o;

		pluginUtil.addCommonOptions(s);

		o = s.option(form.Flag, 'ExtraItems', _('Extra items'),
			_('More details about frequency usage and transitions'));
		o.depends('enable', '1');
	},

	configSummary: function(section) {
		return (section.ExtraItems == '1')
			? _('Detailled CPU frequency monitoring enabled')
			: _('Simple CPU frequency monitoring enabled');
	}
});
